src/Symfony/Bundle/FrameworkBundle/Controller/ProxyController.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Controller;
+
+use Symfony\Component\HttpClient\Exception\InvalidArgumentException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * Proxies a request to a remote URL.
+ *
+ * @author Jérôme Tamarelle <jerome@tamarelle.net>
+ *
+ * @final
+ */
+class ProxyController
+{
+    /** @var HttpClientInterface */
+    private $httpClient;
+
+    public function __construct(HttpClientInterface $httpClient)
+    {
+        $this->httpClient = $httpClient;
+    }
+
+    /**
+     * Creates a response by fetching the content from a remote URL.
+     *
+     * @param Request $request         The request instance
+     * @param string  $url             The URL to fetch
+     * @param string  $method          The HTTP method to use
+     * @param array   $options         Options passed to the HttpClient
+     * @param array   $responseHeaders HTTP headers added to the response
+     *
+     * @throws HttpException In case the route name is empty
+     */
+    public function proxyAction(Request $request, string $url, string $method = 'GET', $options = [], $responseHeaders = []): Response
```

Comment:
If you want to configure routes for this controller with …
```YAML
some_route:
    controller: Symfony\Bundle\FrameworkBundle\Controller\ProxyController
```
… this method should be renamed to `__invoke`. Otherwise, the configuration would look like this:
```YAML
some_route:
    controller: Symfony\Bundle\FrameworkBundle\Controller\ProxyController::proxyAction
```

Alternatively, you can also add an `__invoke` method with the same signature that calls `proxyAction()`.

Reply:
I've opted for `__invoke` only. We don't need many ways of doing the same thing.